File: problems/string-format/string-format.py
import re

def formatter_brute(string, *args):
    for i, argument in enumerate(args):
        string = re.sub("\{" + str(i) + "\}", argument, string)
    return string

def formatter(string, *args):
    #tags = re.findall( r'\{(\d)\}', string)
    for word in string.split():
        print word
    #for i, argument in enumerate(args):
    #    string = re.sub("\{" + str(i) + "\}", argument, string)
    #return string

print formatter('Hello {0} {1}', 'Mr.', 'X')
print formatter('{1}_{0}', '{1}', '{0}')

File: problems/string-format/string-format.rs
fn format_manual(orig_str: &str, args: Vec<&str>) -> String {
    let mut new_str = orig_str.to_string();

    for (i, arg) in args.iter().enumerate() {
        let temp_str = new_str.clone();
        let parts: Vec<&str> = temp_str
            .split(format!("{}{}{}", "{", i.to_string(), "}").as_str())
            .collect();

        new_str = parts.join(arg);
    }

    new_str
}

fn main() {
    println!("{}\n", format_manual("Hello {0} {1}", vec!["Mr.", "X"]));

    println!("{}\n", format!("Hello {0} {1}", "Mr.", "X"));
}
